cloud/pkg/edgecontroller/controller/upstream.go

```diff
@@ -1254,5 +1267,64 @@ func NewUpstreamController(factory k8sinformer.SharedInformerFactory) (*Upstream
 	uc.ruleStatusChan = make(chan model.Message, config.Config.Buffer.UpdateNodeStatus)
 	uc.edgeClusterStateChan = make(chan model.Message, config.Config.Buffer.UpdateEdgeClusterState)
 	uc.missionStateChan = make(chan model.Message, config.Config.Buffer.UpdateMissionState)
+	uc.clusterGatewayChan = make(chan model.Message, config.Config.Buffer.UpdateClusterGateway)
 	return uc, nil
 }
+
+func (uc *UpstreamController) updateClusterGatewayNeighbors() {
+	for {
+		select {
+		case <-beehiveContext.Done():
+			klog.Warning("stop updateClusterGatewayNeighbors")
+			return
+		case msg := <-uc.clusterGatewayChan:
+			klog.V(3).Infof("upstream gateway message %v operation is: %s, and resource is %s", msg, msg.GetOperation(), msg.GetResource())
+
+			namespace, err := messagelayer.GetNamespace(msg)
+			if err != nil {
+				klog.Warningf("process failure, get namespace failed with error: %s", err)
+				continue
+			}
+			name, err := messagelayer.GetResourceName(msg)
+			if err != nil {
+				klog.Warningf("process failure, get resource name failed with error: %s", err)
+				continue
+			}
+			data, err := msg.GetContentData()
+			if err != nil {
+				klog.Warningf("process failure, get content data failed with error: %s", err)
+				return
+			}
+
+			var subClusterGatewayConfigMap v1.ConfigMap
+			if err := json.Unmarshal(data, &subClusterGatewayConfigMap); err != nil {
+				klog.Warningf("failed to get cluster gateway from msg, cluster namesapce: %s, cluster name: %s", namespace, name)
+				return
+			}
+
+			neighborName := subClusterGatewayConfigMap.Data[common.ClusterGatewayConfigMapClusterName]
+			neighborHostIP := subClusterGatewayConfigMap.Data[common.ClusterGatewayConfigMapClusterHostIP]
+			klog.V(3).Infof("upstream gateway neighborName %v, neighborHostIP %v", neighborName, neighborHostIP)
+			configMap, err := uc.kubeClient.CoreV1().ConfigMaps(namespace).Get(context.Background(), common.ClusterGatewayConfigMap, metaV1.GetOptions{})
+			if err != nil {
+				klog.Warningf("Failed to get configmap, namespace: %s, name: %s, err: %v", namespace, name, err)
+			}
+			klog.V(3).Infof("upstream gateway configmap %v", configMap)
+			neighbors := ""
+			if configMap.Data != nil {
+				neighbors = configMap.Data["gateway_neighbors"]
+			}
+			if updatedNeighbors, updated, err := util.GetUpdatedClusterGatewayNeighbors(neighborName, neighborHostIP, neighbors); updated && err == nil {
+				configMap.Data["gateway_neighbors"] = updatedNeighbors
+				updatedConfigMap, err := uc.kubeClient.CoreV1().ConfigMaps(namespace).Update(context.Background(), configMap, metaV1.UpdateOptions{})
+				if err != nil {
+					klog.Warningf("Failed to updated neighbors, namespace: %s, name: %s, err: %v", namespace, name, err)
+					continue
+				}
+				klog.V(3).Infof("upstream gateway configmap updated %v", updatedConfigMap)
```

Comment:
some logs are starting with lower case letters while others are upper cases (line 1321 for example).

Reply:
Updated